### PR TITLE
feat(utility): implement "LazyList.extend"

### PR DESCRIPTION
**Pull request by linjiX**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2epmee6 实现 LazyList.extend 方法](https://app.clickup.com/t/2epmee6)

File: graviti/utility/lazy2.py [hunk 1]
```diff
@@ -26,27 +75,37 @@ def __init__(
...
-        div, mod = divmod(index, self._limit)
-        return self._pages[div][mod]
+        i, j = self._offsets.get_coordinate(index)
+        return self._pages[i][j]
+
+    def extend(self, values: Iterable[Any]) -> None:
+        """Extend mutable sequence by appending elements from the iterable.
+
+        Arguments:
+            values: Elements to be Extended into the mutable sequence.
```
Comment (Lee-000) [inline graviti/utility/lazy2.py:103]: `extended`?

File: graviti/utility/lazy2.py [hunk 2]
```diff
@@ -26,27 +75,37 @@ def __init__(
...
         return self._getitem(index)
 
     def _getitem(self, index: int) -> Any:
-        div, mod = divmod(index, self._limit)
-        return self._pages[div][mod]
+        i, j = self._offsets.get_coordinate(index)
+        return self._pages[i][j]
+
+    def extend(self, values: Iterable[Any]) -> None:
+        """Extend mutable sequence by appending elements from the iterable.
```
Comment (Lee-000) [inline graviti/utility/lazy2.py:100]: `Extend the LazyList by ...`?

File: graviti/utility/lazy2.py [hunk 3]
```diff
@@ -5,13 +5,62 @@
...
+            index: The index of the element in lazy list.
+
+        Returns:
+            The page number and the index of the page.
+
+        """
+        if not hasattr(self, "_offsets"):
+            return divmod(index, self._limit)
+
+        i = bisect_right(self._offsets, index) - 1
```
[the remark this answers is not in the document]
Comment (linjiX) [reply graviti/utility/lazy2.py:47]: check this:
```python
In [19]: offset = [0, 3, 6, 9, 10]

In [20]: bisect_left(offset, 2)
Out[20]: 1

In [21]: bisect_right(offset, 2)
Out[21]: 1
```

File: graviti/utility/lazy2.py [hunk 4]
```diff
@@ -26,27 +75,37 @@ def __init__(
...
-        div, mod = divmod(index, self._limit)
-        return self._pages[div][mod]
+        i, j = self._offsets.get_coordinate(index)
+        return self._pages[i][j]
+
+    def extend(self, values: Iterable[Any]) -> None:
+        """Extend LazyList by appending elements from the iterable.
+
+        Arguments:
+            values: Elements to be extended into the mutable sequence.
```
Comment (Lee-000) [inline graviti/utility/lazy2.py:103]: `into the LazyList`?

**Review by Lee-000 — APPROVED** [review]

**Review by AChenQ — APPROVED** [review]

**Pull request merged**